### DISCUSSION TITLE
Black bars top/and bottom on android

### DISCUSSION THREAD

**Discussion by raresn** [airsdk/Adobe-Runtime-Support#1394] [Q&A] [answered]
Hey guys!
Our app doesn't stretch out properly on android phones, we have some extra black space top/bottom.
How can we get rid of these?
If we go to settings, we can add the app under "full screen apps", and that solves it, but it is something we wanna flag automaticaly. 
![image](https://user-images.githubusercontent.com/1768694/143580621-af1189b2-c05c-4999-86c1-36398bb29c95.png)

Thank you!

**Comment by MalacTheLittle** [comment, marked as answer]
You should add `hardwareAcceleration` and `max_aspect` support to your app descriptor:

```
<android>
    <manifestAdditions><![CDATA[<manifest>
        <application android:hardwareAccelerated="true">
	    <meta-data android:name="android.max_aspect" android:value="2.16" />
        </application>
        </manifest>]]>
    </manifestAdditions>
</android>

```

**Reply by marchbold** [reply]
Just was writing this exact thing :) Would suggest increasing the max_aspect to 2.5 though.

**Reply by MalacTheLittle** [reply]
@marchbold Can you please explain what does this number means? Thanks :)

**Reply by marchbold** [reply]
Just sets the maximum aspect ratio that your application supports, Android will add black bars for any device that is larger than this ratio. The default is smaller so it causes these black bars on many of the modern android devices with high aspect ratios.

**Reply by ajwfrost** [reply]
... makes me wonder whether we should be settings something like 2.5 as the default already. Not sure what the downside would be to this (or why Android has such a low default value...) ..?

**Reply by MalacTheLittle** [reply]
Perfect! Thanks! Then I'll have to update my apps :)

@raresn FYI, you can also stretch your app under cutout for devices with cutout front camera by adding `notch_support` also under `hardwareAcceleration`:
```
<meta-data android:name="android.notch_support" android:value="true"/>
```
The only problem is that there's still no way to get height of cutout unless you use ANE.

**Reply by raresn** [reply]
@MalacTheLittle  - good one - thanks for the tip!

**Reply by MalacTheLittle** [reply]
Just a little update... to support notch on Xiaomi phone models add next line:

`<meta-data android:name="notch.config" android:value="portrait|landscape"/>`

I've tested it on Xiaomi 12 Lite (1080×2400) and with next manifest additions it works like a charm:

```
<application android:hardwareAccelerated="true">
	<meta-data android:name="android.max_aspect" android:value="2.5"/>
	<meta-data android:name="android.notch_support" android:value="true"/>
	<meta-data android:name="notch.config" android:value="portrait|landscape"/>
</application>
```
Also, with new [Screen.mainScreen.safeArea](https://airsdk.dev/reference/actionscript/3.0/flash/display/Screen.html#safeArea) property you can get notch height. Go Harman 💪

**Comment by raresn** [comment]
Ha! Thanks guys! I will give it a spin!
